velox/functions/sparksql/Arithmetic.h

```diff
@@ -606,4 +606,44 @@ struct CheckedDivideFunction {
     return Status::OK();
   }
 };
+
+template <typename T>
+struct FactorialFunction {
```

Comment:
Would you add documentation for this function in https://github.com/facebookincubator/velox/blob/main/velox/docs/functions/spark/math.rst?